src/tests/rpp/test_finally.cpp

```diff
@@ -0,0 +1,115 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#include <snitch/snitch.hpp>
+
+#include <rpp/operators/finally.hpp>
+#include <rpp/sources/create.hpp>
+
+#include "disposable_observable.hpp"
+#include "mock_observer.hpp"
+
+TEST_CASE("finally executes only at the end")
+{
+    auto mock = mock_observer_strategy<int>();
+    SECTION("observable with no emissions")
+    {
+        auto   obs     = rpp::source::create<int>([](const auto&) {
+        });
+        size_t invoked = 0;
+        SECTION("subscribe")
+        {
+            obs | rpp::operators::finally([&]() noexcept { ++invoked; })
```

Comment:
@victimsnino do you think we can relax the noexcept requirement?

Reply:
Theoretically we can, but then we should handle it somehow properly. In my understanding it is not possible. Reason: dispose could happen from ANY place and we don't know where exception should be propogated. Also it could break internal disposable state....

Reply:
I see, I added a comment in doc to specify it